internal/control_plane/id_generator/idgenerator.go

```diff
@@ -0,0 +1,15 @@
+package id_generator
+
+import (
+	id "github.com/sony/sonyflake"
+	"octavius/internal/control_plane/logger"
+)
+
+func NextID() uint64 {
+	var idGenerator = id.NewSonyflake(id.Settings{})
+	uid, err := idGenerator.NextID()
+	if err != nil {
+		logger.Error(err, "unique id generation failed")
```

Comment:
return the error.

Reply:
When we are logging it right way, is it necessary to return it too, I mean returning the error will add error check in the service.go file and we will do the same thing (error logging) twice please CMIIW.

Reply:
in go, the error should only be returned **OR** logged. never do both, the one that will do the logging is the top-level caller which in your case the `service.go` this function will only return the error. or you can do error wrapping to add more context.

Reply:
Yeah! top-level caller thing makes sense. Updated